### PR TITLE
Add tests for `dev` and `bundle` commands

### PR DESCRIPTION
**Pull request by thecb4**
In support of handling #99 

Added
* carton dev test (no arguments)
* carton bundle test (no arguments)

### CODE REVIEW

**Review by MaxDesiatov — DISMISSED** [review]
LGTM 👍

**Review by yonihemi — DISMISSED** [review]

**Comment by thecb4** [comment]
@MaxDesiatov @yonihemi FYI. The Bundle test is behaving strangely. I am using the CI to debug since all the tests pass on my machine and in the docker_image test.

**Comment by thecb4** [comment]
@MaxDesiatov @yonihemi 

Looks like an executable is not being installed  **'wasm-opt'**

```
Test Case 'BundleCommandTests.testWithNoArguments' started at 2020-12-27 17:31:31.150
996
Directory doesn't exist or contains outdated polyfills: /home/runner/.carton/static
997
Polyfills archive successfully downloaded from https://github.com/swiftwasm/carton/releases/download/0.8.2/static.zip
998
Unpacking the archive: /home/runner/.carton/static.zip
999
Entrypoint integrity verified: /home/runner/.carton/static/bundle.js
1000
- checking Swift compiler path: /home/runner/.carton/sdk/wasm-5.3.1-RELEASE/usr/bin/swift
1001
- checking Swift compiler path: /home/runner/.swiftenv/versions/wasm-5.3.1-RELEASE/usr/bin/swift
1002
Fetching release assets from https://api.github.com/repos/swiftwasm/swift/releases/tags/swift-wasm-5.3.1-RELEASE
1003
Response contained body, parsing it now...
1004
Response succesfully parsed, choosing from this number of assets: 3
1005
Local installation of Swift version wasm-5.3.1-RELEASE not found
1006
Swift toolchain/SDK download URL: https://github.com/swiftwasm/swift/releases/download/swift-wasm-5.3.1-RELEASE/swift-wasm-5.3.1-RELEASE-ubuntu18.04_x86_64.tar.gz
1007
Archive size is 680 MB
1008
Downloading the archive
1009
0%: saving to /home/runner/.carton/sdk/wasm-5.3.1-RELEASE.gz
1010
Download completed successfully
1011
Unpacking the archive: tar xzf /home/runner/.carton/sdk/wasm-5.3.1-RELEASE.gz --strip-components=1 --directory /home/runner/.carton/sdk/wasm-5.3.1-RELEASE
1012
- checking Swift compiler path: /home/runner/.carton/sdk/wasm-5.3.1-RELEASE/usr/bin/swift
1013
Inferring basic settings...
1014
- swift executable: /home/runner/.carton/sdk/wasm-5.3.1-RELEASE/usr/bin/swift
1015
SwiftWasm Swift version 5.3 (swiftlang-5.3.1)
1016
Target: x86_64-unknown-linux-gnu
1017

1018
Parsing package manifest: /home/runner/.carton/sdk/wasm-5.3.1-RELEASE/usr/bin/swift package dump-package
1019
- development binary to serve: /home/runner/work/carton/carton/Tests/Fixtures/Milk/.build/wasm32-unknown-wasi/release/Milk.wasm
1020

1021
Building the project before spinning up a server...
1022

1023
Compiling...
1024
/home/runner/.carton/sdk/wasm-5.3.1-RELEASE/usr/bin/swift build -c release --product Milk --enable-test-discovery --triple wasm32-unknown-wasi
1025

1026

1027
`swift` process finished successfully
1028
`swift build` completed in 50.55 seconds
1029
Binary transformation for Safari compatibility completed in 0.29 seconds
1030
Right after building the main binary size is 31.61 MB
1031

1032
After stripping debug info the main binary size is 23.06 MB
1033

1034

1035
wasm-opt -Os /home/runner/work/carton/carton/Tests/Fixtures/Milk/.build/wasm32-unknown-wasi/release/Milk.wasm -o /home/runner/work/carton/carton/Tests/Fixtures/Milk/Bundle/main.wasmRunning...
1036

1037

1038
Process failed and produced following output: 
1039
could not find executable for 'wasm-opt'
```

File: Tests/CartonCommandTests/CommandTestHelper.swift [hunk 1]
```diff
@@ -177,26 +186,23 @@ public func AssertHelp<T: ParsableCommand, U: ParsableCommand>(
...
       : bundleURL
   }
 
-  static func AssertExecuteCommand(
+  func executeCommand(
     command: String,
     cwd: URL? = nil, // To allow for testing of file based output
     expected: String? = nil,
     exitCode: ExitCode = .success,
+    // watcher: @escaping (_ pid: Int32) -> Void,
```
Comment (MaxDesiatov) [inline Tests/CartonCommandTests/CommandTestHelper.swift:202]:
```suggestion
```

**Review by MaxDesiatov — APPROVED** [review]
LGTM 👍

**Pull request merged**